examples/perf/ipc.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/python
+# IPC - Instructions Per Cycles using Perf Events and
+# uprobes
+# 24-Apr-2020	Saleem Ahmad	Created this.
+
+from __future__ import print_function
+from bcc import BPF, USDT, Perf
+from bcc.utils import printb
+from time import sleep
+import sys
+from optparse import OptionParser
+
+# load BPF program
+code="""
+#include <uapi/linux/ptrace.h>
+
+struct perf_delta {
+    u64 clk_delta;
+    u64 inst_delta;
+    u64 time_delta;
+};
+
+/*
+Perf Arrays to read counter values for open
+perf events.
+BPF_PERF_ARRAY taken maximum number of cpus.
+Change it to the max number of cpu in case
+it is greater than MAX_CPUS
+*/
+#define MAX_CPUS 128
+BPF_PERF_ARRAY(clk, MAX_CPUS);
```

Comment:
We could potentially [pass the number of CPUs](https://github.com/iovisor/bcc/blob/98c18b640117b10f923c8697be8bfff8ad39834b/src/python/bcc/utils.py#L35) through [Cflags list](https://github.com/iovisor/bcc/blob/master/src/python/bcc/__init__.py#L286), this way it should work in all systems and does not require manually updates :D

Reply:
Changed to pass CPU number from cflags